bots/pr/src/main/java/org/openjdk/skara/bots/pr/BranchCommand.java

```diff
@@ -0,0 +1,133 @@
+                        + " and your GitHub account needs to be linked with your OpenJDK username"
+                        + " ([how to associate your GitHub account with your OpenJDK username]"
+                        + "(https://wiki.openjdk.org/display/skara#Skara-AssociatingyourGitHubaccountandyourOpenJDKusername)).");
+                return;
+            }
+
+            var args = command.args();
+            if (args.isBlank()) {
+                showHelp(reply);
+                return;
+            }
+
+            var parts = args.split(" ");
+            if (parts.length > 1) {
+                showHelp(reply);
+                return;
+            }
+            var branchName = parts[0];
+
+            var localRepoDir = scratchArea.get(this)
+                    .resolve(bot.repo().name());
+            var localRepo = bot.hostedRepositoryPool()
+                               .orElseThrow(() -> new IllegalStateException("Missing repository pool for PR bot"))
+                               .materialize(bot.repo(), localRepoDir);
+            localRepo.fetch(bot.repo().authenticatedUrl(), commit.hash().toString(), true);
+
+            var remoteBranches = localRepo.remoteBranches(bot.repo().authenticatedUrl().toString());
+            var remoteBranchNames = remoteBranches.stream()
+                                                  .map(Reference::name)
+                                                  .collect(Collectors.toSet());
+            if (remoteBranchNames.contains(branchName)) {
+                var ref = remoteBranches.stream().filter(r -> r.name().equals(branchName)).findFirst();
+                if (ref.isEmpty()) {
+                    throw new IllegalStateException("Cannot find hash for remote branch '" + branchName + "'");
+                }
+                var hash = ref.get().hash();
+                var hashUrl = bot.repo().webUrl(hash);
+                reply.println("A branch with name `" + branchName + "` already exists that refers to commit [" + hash.abbreviate() + "](" + hashUrl + ").");
+                return;
+            }
+
+            var jcheckConf = JCheckConfiguration.from(localRepo, commit.hash());
+            var branchPattern = jcheckConf.isPresent() ? jcheckConf.get().repository().branches() : null;
+            if (branchPattern != null && !branchName.matches(branchPattern)) {
+                reply.println("The given branch name `" + branchName + "` is not of the form `" + branchPattern + "`.");
+                return;
+            }
+
+            var branch = localRepo.branch(commit.hash(), branchName);
+            log.info("Pushing branch '" + branch + "' to refer to commit: " + commit.hash().hex());
+            localRepo.push(commit.hash(), bot.repo().authenticatedUrl(), branch.name(), false, false);
+            log.info(String.join(", ", localRepo.branches().stream().map(Branch::name).collect(Collectors.toList())));
```

Comment:
This will potentially be a rather long log message. Not sure we need to list all branches in the repo.

Reply:
Yeah, sorry, this was a left-over debug message.